modules/javafx.graphics/src/test/java/test/javafx/scene/paint/RadialGradientTest.java

```diff
@@ -284,6 +284,22 @@ public void testValueOfIllegalNotations() {
         } catch (IllegalArgumentException iae) {
             // expected
         }
+
+        // radius can't be negative
+        try {
+            RadialGradient.valueOf("radial-gradient(radius -100, red 0%, blue 30%, black 100%)");
+            fail("IllegalArgument should have been thrown.");
+        } catch (IllegalArgumentException iae) {
+            assertTrue(iae.getMessage().contains("radius cannot be negative"));
+        }
+
+        // focusDistance can't be negative
+        try {
+            RadialGradient.valueOf("radial-gradient(focus-distance -100%, red 0%, blue 30%, black 100%)");
+            fail("IllegalArgument should have been thrown.");
+        } catch (IllegalArgumentException iae) {
+            assertTrue(iae.getMessage().contains("focus-distance cannot be negative"));
+        }
```

Comment:
`RadialGradient` adds `checkInvariants` in three places, but only two tests are added.
Should we add one more?

Reply:
No, but I've removed the `checkInvariants()` call in the private constructor, as it is only called from the `interpolate()` method and thus always operates with valid gradients (no need to verify what has already been verified).